jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +165,46 @@ public void removeComment(final String serverTemplateId, final String containerI
         return comparing(CaseMilestoneSummary::getStatus).thenComparing(request.getSortByAsc() ? comparatorByName: comparatorByName.reversed());
     }
 
+    @Override
+    public CaseActions getCaseActions(String containerId, String caseId, String userId) {
+        List<String> allStatus = new ArrayList<>();
+        allStatus.addAll(CaseActionStatus.AVAILABLE.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.IN_PROGRESS.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_FAIL.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_SUCCESS.getTaskStatuses());
+
+        List<TaskSummary> actionSummaries = client.findCaseTasksAssignedAsBusinessAdministrator(caseId, userId, allStatus, 0, PAGE_SIZE_UNLIMITED, "createdOn", false);
+        List<CaseActionSummary> allActionsSummaries = actionSummaries.stream().map(new CaseActionMapper()).collect(toList());
+        List<CaseActionSummary> availableActions = getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.AVAILABLE);
+        availableActions.addAll(client.getAdHocFragments(containerId, caseId)
+                .stream()
+                .map(new CaseActionAdHocMapper())
+                .sorted(comparing(CaseActionSummary::getName)).collect(toList()));
+
+
+        CaseActions caseActions = new CaseActions();
+        caseActions.setAvailableActions(availableActions);
+        caseActions.setInProgressAction(getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.IN_PROGRESS));
+        caseActions.setCompleteActions(getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.COMPLETED_FAIL, CaseActionStatus.COMPLETED_SUCCESS));
+        return caseActions;
+    }
+
+    protected List<CaseActionSummary> getCaseActionsByStatus(List<CaseActionSummary> actionSummaries, CaseActionStatus... caseActionStatuses) {
+        List<String> statusList = new ArrayList<>();
+        for (CaseActionStatus status : caseActionStatuses) {
+            statusList.addAll(status.getTaskStatuses());
+        }
+        return actionSummaries.stream()
+                .filter(c -> statusList.contains(c.getStatus())).collect(toList());
+    }
+
+    public void addDynamicUserTask(String containerId, String caseId, String name, String description, String actors, String groups, Map<String, Object> data) {
+        client.addDynamicUserTask(containerId, caseId, name, description, actors, groups, data);
+    }
+
+    @Override
+    public void triggerAdHocActionInStage(String containerId, String caseId, String stageId, String adHocName, Map<String, Object> data) {
+        client.triggerAdHocFragmentInStage(containerId, caseId, stageId, adHocName, data);
```

Comment:
It looks like stageId argument hasn't been used anywhere in the code, so I'm wondering why has this method been chosen over triggerAdHocFragment method (which doesn't consider case stages)?